src/tasks/block/sim.rs

```diff
@@ -265,13 +172,14 @@ impl Simulator {
     /// An `Instant` representing the simulation deadline, as calculated by determining
     /// the time left in the current slot and adding that to the current timestamp in UNIX seconds.
     pub fn calculate_deadline(&self) -> Instant {
-        // Calculate the current timestamp in seconds since the UNIX epoch
-        let now = SystemTime::now();
-        let unix_seconds = now.duration_since(UNIX_EPOCH).expect("Time went backwards").as_secs();
-        // Calculate the time remaining in the current slot
-        let remaining = self.slot_calculator.calculate_timepoint_within_slot(unix_seconds);
-        //  Deadline is equal to the start of the next slot plus the time remaining in this slot
-        Instant::now() + Duration::from_secs(remaining)
```

Comment:
this seems to have been giving incorrect output. we need to sub remaining from duration, then add that to now